libshaderc_spvc/src/spvc_private.cc

```diff
@@ -288,14 +289,47 @@ shaderc_spvc_compilation_result_t generate_hlsl_shader(
     const uint32_t* source, size_t source_len,
     shaderc_spvc_compile_options_t options,
     shaderc_spvc_compilation_result_t result) {
-  std::unique_ptr<spirv_cross::CompilerHLSL> compiler(
-      new (std::nothrow) spirv_cross::CompilerHLSL(source, source_len));
-  if (!compiler) {
-    result->messages.append(
-        "Unable to initialize SPIRV-Cross HLSL compiler.\n");
+
+  std::unique_ptr<spirv_cross::CompilerHLSL> compiler;
+
+  // spvc is under development, for now run spirv-cross compiler by default
+  // TODO (sarahM0): change the default to spvc when it's done
+  #if ENABLE_SPVC_PARSER
+    spvtools::Optimizer opt(options->source_env);
+    opt.SetMessageConsumer(std::bind(
+        consume_spirv_tools_message, result, std::placeholders::_1,
+        std::placeholders::_2, std::placeholders::_3, std::placeholders::_4));
+
+    spirv_cross::ParsedIR ir;
+    ir.spirv = std::vector<uint32_t>(source, source + source_len);
+
+    opt.RegisterPass(
+        spvtools::Optimizer::PassToken(std::unique_ptr<spvtools::opt::Pass>(
+            reinterpret_cast<spvtools::opt::Pass*>(
+                new spvtools::opt::SpvcIrPass(ir)))));
+
+    if (!opt.Run(source, source_len, &result->binary_output)) {
+      result->messages.append(
+          "Transformations between source and target "
+          "execution environments failed.\n");
+      result->status = shaderc_compilation_status_transformation_error;
+      compiler.reset(new (std::nothrow) spirv_cross::CompilerHLSL(ir));
+    }
+
+    result->messages.append("Tried to do a temporary pass.\n");
```

Comment:
nit: Have this string be clearer about what is occuring, something like "Expected failure due to incomplete spvc IR generation implementation."